packages/wrangler/src/deploy/index.ts

```diff
@@ -254,6 +254,12 @@ export async function deployHandler(args: DeployArgs) {
 		args.config || (args.script && findWranglerToml(path.dirname(args.script)));
 	const projectRoot = configPath && path.dirname(configPath);
 	const config = readConfig(configPath, args);
+	if (config.pages_build_output_dir) {
+		throw new UserError(
+			"This command is for Workers, for Pages please run `wrangler pages deploy`."
```

Comment:
```suggestion
			"You are attempting to run a Workers specific command in a Pages project. Please use `wrangler pages deploy` instead."
```

or alternatively

```suggestion
			"You are attempting to run `wrangler deploy` in a Pages project. `wrangler deploy` is meant to deploy Workers to Cloudflare. For Pages, please use `wrangler pages deploy` instead."
```

or something along those lines

Reply:
same for all the other errors